src/components/can_cooperation/src/request_controller.cc

```diff
@@ -42,16 +42,11 @@ namespace request_controller {
 CREATE_LOGGERPTR_GLOBAL(logger_, "CANRequestController")
 
 RequestController::RequestController() {
-  // TODO(PV): move setting to separate instance
-  functional_modules::TimeUnit timeout_seconds = 10;
-  std::ifstream in("./plugins/can_config.json");
-  if (in.is_open()) {
-    Json::Reader reader;
-    Json::Value value;
-    if (reader.parse(in, value, false)) {
-      timeout_seconds = value["timeout_period_seconds"].asUInt();
-    }
-  }
+  functional_modules::TimeUnit timeout_seconds = 100;
+  functional_modules::Settings settings;
+  settings.ReadParameter<functional_modules::TimeUnit>("Remote Control",
```

Comment:
You can write settings.ReadParameter("Remote Control", "timeout_period_seconds", &timeout_seconds)
Isn't it?